src/realizations/catchment/Tshirt_C_Realization.cpp

```diff
@@ -434,7 +432,25 @@ double Tshirt_C_Realization::get_response(time_step_t t_index, time_step_t t_del
     //  forcing.  It may actually belong within the forcing object.
 
     // TODO: it also needs to account for getting the right precip data point (i.e., t_index may not be "next")
-    double precip = this->legacy_forcing->get_next_hourly_precipitation_meters_per_second();
+    //Checking the time step used is consistent with that provided in forcing data
+    time_t t_delta = this->forcing->record_duration();
+    if (t_delta != t_delta_s) {    //Checking the time step used is consistent with that provided in forcing data
+        throw std::invalid_argument("Getting response using insonsistent time step with provided forcing data");
+    }
+    //Negative t_index is not allowed
+    if (t_index < 0) {
+        throw std::invalid_argument("Getting response of negative time step in Tshirt C Realization is not allowed.");
+    }
+    time_t start_time = this->forcing->get_data_start_time();
```

Comment:
While `time_t` is probably reasonable here, I know in some other places we have used `long` because we may deal with dates < 1970. Is this based on code in another module?

Reply:
time_t is used all over the places in our codes. In CsvPerFeatureForcingProvider.hpp, time_t is used for epoch time. This is used here for consistency. If you think that might not be enough, we can change that.